### DISCUSSION TITLE
Train and test set are merged in experiment setup

### DISCUSSION THREAD

**Discussion by TuanDTr** [pycaret/pycaret#3525] [General]
Hi all,

I have a simple setup to use the same dataset for training and testing. 
```python
from sklearn.datasets import load_breast_cancer 
from pycaret.classification import ClassificationExperiment

X, y = load_breast_cancer(return_X_y=True, as_frame=True)
X["target"] = y

s = ClassificationExperiment()
s.setup(data=X, target="target", test_data=X)

print("Size of original data", X.shape)
print("Size of train set in pycaret", s.train.shape)
print("Size of test set in pycaret", s.test.shape)
```
Output:
```
Size of original data (569, 31)
Size of train set in pycaret (1138, 31)
Size of test set in pycaret (1138, 31)
```
What I get from the output is that Pycaret combines train and test set into one single dataframe. I think this is intentional but I don't know the purpose of it. Is anyone aware of this?
